pulser/simulation/simulation.py

```diff
@@ -15,116 +15,266 @@
-                 evaluation_times: Union[str, ArrayLike] = 'Full') -> None:
-        """Initialize the Simulation with a specific pulser.Sequence."""
-        supported_bases = {"ground-rydberg", "digital"}
+                 config: Optional[SimConfig] = None,
+                 evaluation_times: Union[float, str, ArrayLike] = 'Full'
+                 ) -> None:
+        """Instantiates a Simulation object."""
         if not isinstance(sequence, Sequence):
             raise TypeError("The provided sequence has to be a valid "
                             "pulser.Sequence instance.")
         if not sequence._schedule:
             raise ValueError("The provided sequence has no declared channels.")
+        if all(sequence._schedule[x][-1].tf == 0 for x in sequence._channels):
+            raise ValueError("No instructions given for the channels in the "
+                             "sequence.")
         not_supported = (set(ch.basis for ch in sequence._channels.values())
-                         - supported_bases)
+                         - SUPPORTED_BASES)
         if not_supported:
             raise NotImplementedError("Sequence with unsupported bases: "
                                       + "".join(not_supported))
-        if all(sequence._schedule[x][-1].tf == 0 for x in sequence._channels):
-            raise ValueError("No instructions given for the channels in the "
-                             "sequence.")
         self._seq = sequence
         self._qdict = self._seq.qubit_info
         self._size = len(self._qdict)
         self._tot_duration = max(
             [self._seq._last(ch).tf for ch in self._seq._schedule]
         )
-
         if not (0 < sampling_rate <= 1.0):
             raise ValueError("The sampling rate (`sampling_rate` = "
                              f"{sampling_rate}) must be greater than 0 and "
                              "less than or equal to 1.")
-        if int(self._tot_duration*sampling_rate) < 4:
+        if int(self._tot_duration * sampling_rate) < 4:
             raise ValueError("`sampling_rate` is too small, less than 4 data "
                              "points.")
-        self.sampling_rate = sampling_rate
-
+        self._sampling_rate = sampling_rate
         self._qid_index = {qid: i for i, qid in enumerate(self._qdict)}
-        self.samples: dict[str, dict[str, dict]] = {
-            addr: {basis: {} for basis in ['ground-rydberg', 'digital']}
-            for addr in ['Global', 'Local']
-        }
-        self.operators = deepcopy(self.samples)
+        self._collapse_ops: list[qutip.Qobj] = []
+        self._times = self._adapt_to_sampling_rate(
+            np.arange(self._tot_duration, dtype=np.double)/1000)
+        self.evaluation_times = evaluation_times
+        self._bad_atoms: dict[Union[str, int], bool] = {
+            qid: False for qid in self._qid_index}
+        self._doppler_detune: dict[Union[str, int], float] = {
+            qid: 0. for qid in self._qid_index}
```

Comment:
Actually, since we initialize these inside `set_config`, why not declare them just as empty dicts here?
```suggestion
        self._bad_atoms: dict[Union[str, int], bool] = {}
        self._doppler_detune: dict[Union[str, int], float] = {}
```